lib/access-context.js

```diff
@@ -126,8 +128,20 @@ AccessContext.prototype.addPrincipal = function(principalType, principalId, prin
 AccessContext.prototype.getUserId = function() {
   for (var i = 0; i < this.principals.length; i++) {
     var p = this.principals[i];
+    // the principalType must either be 'USER' or permit to resolve a valid user model
     if (p.type === Principal.USER) {
       return p.id;
+    } else {
+      if (!this.registry) return null;
```

Comment:
> `principals` contains "APP", "CustomUser", "USER" - returns id of the second principal

 I suggest skipping this test (and the similar one `principals` contains "APP", "USER", "CustomUser".
If CustomUser is a valid user model then the developper should have defined the expected polymorphic relation with AccessToken and then the principalType 'USER' should be meaningless. The next one your proposed will cover edge cases